File: uva573.py
#/usr/bin/env python3

if __name__ == "__main__":
    while True:
        H,U,D,F = map(int, input().strip().split())
        if H == 0:
            break

        F = U*F/100.0
        height = 0
        
        i = 0

        while True:
            i += 1
            height += U
            if height > H:
                print("success on day ", i)
                break

            height -= D

            if height < 0:
                print("failure on day ", i)
                break
            U -= F
            if U <0:
                U = 0

File: uva573.cpp
#include<iostream>
#include<cstdio>
using namespace std;

int main() {
    while(true) {
        int H;
        double  U, D, F;
        cin >> H >> U >> D >> F;
        if (H == 0)
            break;

        F = U * F/100.0;
        int i = 1;
        double height = U;
        while(true) {
            if( height > H) {
                printf("success on day %d\n", i);
                break;
            } 
            if (height <= 0) {
                printf("failure on day %d\n", i);
                break;
            }

            height -= D;
            U -= F;
            if (U < 0)
                U=0;
            height += U;
            i++;
            cout << height << endl;
        }
    }
}
